### ISSUE TITLE
the difference of Energy and force Error for the training and validation set is big

### ISSUE THREAD

**Issue by Manyi-Yang** [atomistic-machine-learning/schnetpack#338]
Hi, When did some tests using the SchNet package, I got the following results.  

### Energy and force Error for the training and validation set
![image](https://user-images.githubusercontent.com/50021753/134016792-0a2f51e0-8af5-42ec-bafd-408c54d54fc3.png)

> train_dataset MAE:
    energy:    0.30023 meV/atom
    forces:  158.48391 meV/Å
   val_dataset MAE:
    energy:    2.45322 meV/atom
    forces:  171.90308 meV/Å

Here I have some questions:
1. the energy error for the validation set is about 8 times larger than that of the training set (very good). Is this normal? What are the possible reasons for this result?
2. the force error for both the training set and the validation set is quite larger,  this is a little strange for me because at least the energy error for the training set is quite good. 
3. Another question:  can you give me some suggestions on how to choose the value of `batch_size`

 Please give me suggestions on how to improve the model. Thank you very much! 
### Here is my args.json file
> {
    "aggregation_mode": "sum",
    "batch_size": 2,
    "checkpoint_interval": 1,
    "contributions": null,
    "cuda": true,
    "cutoff": 8.0,
    "cutoff_function": "cosine",
    "datapath": "/home/myang@iit.local/S/SchNet/Only_S8/dpgen_raw_data_to_i9/aimd_dpgen_raw/model/dataset.db",
    "dataset": "custom",
    "derivative": "forces",
    "environment_provider": "torch",
    "features": 48,
    "force": null,
    "interactions": 6,
    "keep_n_checkpoints": 3,
    "log_every_n_epochs": 1,
    "logger": "csv",
    "lr": 0.0005,
    "lr_decay": 0.8,
    "lr_min": 1e-06,
    "lr_patience": 25,
    "max_epochs": 10000,
    "max_steps": null,
    "mode": "train",
    "model": "schnet",
    "modelpath": "model/",
    "n_epochs": 100000,
    "n_layers": 3,
    "negative_dr": true,
    "normalize_filter": false,
    "num_gaussians": 25,
    "output_module": "atomwise",
    "overwrite": false,
    "parallel": true,
    "property": "energy",
    "rho": 0.1,
    "seed": null,
    "split": [
        2100,
        124
    ],
    "split_path": null,
    "stress": null
}

**Comment by Stefaanhess** [comment]
Hey @Manyi-Yang ,
this looks like your model is overfitting. This can be related to the rather small set of training points in your dataset. If you can not increase the dataset, you could still try to use a smaller model. I would maybe try to change the following settings: `n_interactions=3, n_features=64, n_layers=1`. (Question1)
Regarding question 2: This could also be related to overfitting on the energy labels. You could change the loss-tradeoff and see if this helps. Therefore you can change `rho`. Maybe you could try the following values: `rho=0` (training on forces only), `rho=1` (training on energy only) and `rho=0.01` (stronger weight towards forces). Aftwerwards you can compare the results and see how good a training on only 1 property can get.
I would also recommend a bigger batchsize, because this could help to reduce overfitting. If I remember correctly, you are using big systems, so you will have to check where the memory limits of your hardware are.
If this does not help to solve the problem, you can still try to use the PaiNN model in the dev branch. This will be released and merged to master in some weeks.

**Comment by Manyi-Yang** [comment]
Hey @Stefaanhess 
Thank you very much. 
### Test results
I have tried the parameters that you suggested  `n_interactions=3, n_features=64, n_layers=1 roh=0.01`
I got the following results. It seems the overfitting problem has been solved but the MAE Error for both the energy and force are increased. 
```
Number of points in val_dataset :124
MAE:val_dataset
    energy:    4.08493 meV/atom 
     forces:    193.834 meV/Å 
Number of points in train_dataset :2100
MAE:train_dataset
    energy:    2.38001 meV/atom 
     forces:    178.906 meV/Å
```
I also did the tests that only trained on force `roh=0.0 `and energy` roh=10 ` :  
if` roh=0.0,` the force becomes slightly better. The energy has a shift, which is easy to understand.
if` roh=1.0`, the result is a mess.  it means the force is necessary. 
It seems at the moment it is difficult to improve the accuracy of the force, right?
```
### **roh=0.0** 
Number of points in val_dataset :124
MAE:val_dataset
    energy:  329.24504 meV/atom 
     forces:    178.936 meV/Å 
Number of points in train_dataset :2100
MAE:train_dataset
    energy:  329.58044 meV/atom 
     forces:    169.168 meV/Å 

### **roh=1.0** 
Number of points in val_dataset :124
MAE:val_dataset
    energy:    6.47760 meV/atom 
     forces:    705.361 meV/Å 
Number of points in train_dataset :2100
MAE:train_dataset
    energy:    0.91868 meV/atom 
     forces:    699.444 meV/Å 
```
 Maybe I really need a large set of training points in my dataset. 
Actually,  I also trained a model with about  35K training points in the dataset(coved more complicated configurational space than which I test above), which is super slow. The MEA energy and force of this model are too big. So this is the reason I did the test with only a small part of the configurational space as above. 
Anyway, I will try to use the PaiNN model.  Thanks again.

**Comment by Stefaanhess** [comment]
Hey @Manyi-Yang 
it is hard to tell the optimal settings for your model, but the train-loss being a bit better than the val-loss should be ok.
I would suggest to try if PaiNN works better for you :)

**Comment by anywallsocket** [comment]
As far as I'm aware this is typical behavior for a couple of reasons:
1. forces have 3xa number of data points compared to energies, where a is atoms in your system.
2. you're writing the force units in eV/Å, but it is eV/a/Å, so check "divide_by_atoms" in DataModule if using.
3. 10,000 epocs seems excessive unless you have a truly massive system (in which case batch size should be large)

Try rho=0.5 for even split to start -- looks like you need to preform some hyper parameter sweeps first.
Also your tags say property "energy", but it should say "forces" as well -- check the tutorials for that.

For reference, when I train atomic systems <50 atoms I get MAE energy about 3x MAE forces.
I'd recommend more features less interactions and a smaller cutoff.

Specifically: Look at your training plot -- overtraining is obvious if there is barely any improvement for several epochs.
